### DISCUSSION TITLE
OPEN_INPUT:Invalid data found when processing input

### DISCUSSION THREAD

**Discussion by Fitap** [doitandbedone/ispyagentdvr-docker#541] [General]
Hi,

Any of you use IMOU brand cameras CUE 2 model?
I was unable to get camera works.
This is the log output and I appreciate your help.
Thanks.

18:56:04 | Starting...
18:56:04 | Detected: Linux64
18:56:04 | Using default media location
18:56:04 | Copying /agent/Content/Masks/demo_mask.png
18:56:04 | Copying /agent/Content/sounds/police.wav
18:56:04 | Copying /agent/Content/sounds/alsation_bark.wav
18:56:04 | Copying /agent/Content/sounds/doorbell.wav
18:56:04 | Copying /agent/Content/sounds/alarm.wav
18:56:04 | Copying /agent/Content/sounds/airhorn.wav
18:56:04 | Copying /agent/Content/sounds/alsationxsiberian_bark.wav
18:56:04 | Copying /agent/Content/sounds/cairnterrierbark.wav
18:56:04 | Copying /agent/Content/sounds/rottweiler_bark.wav
18:56:04 | Copying /agent/Content/sounds/pop.wav
18:56:04 | Copying /agent/Content/sounds/wiggle.wav
18:56:04 | Copying /agent/Content/sounds/spanielgrowlbark.wav
18:56:06 | Creating temp path for opencv in /tmp/
18:56:06 | Get com port list
18:56:06 | Agent (v4.4.1.0)
18:56:06 | Log start
18:56:11 | Using WorkStation GC
18:56:11 | Init FFmpeg
18:56:11 | Looking for ffmpeg in ffmpeg-v5/workspace/lib
18:56:11 | Not in there..
18:56:11 | Looking for ffmpeg in /usr/local/lib
18:56:11 | Not in there..
18:56:11 | Looking for ffmpeg in /usr/lib/x86_64-linux-gnu
18:56:11 | Found ffmpeg in /usr/lib/x86_64-linux-gnu
18:56:11 | ffmpeg path: /usr/lib/x86_64-linux-gnu
18:56:11 | Initializing devices
18:56:11 | vdpau
18:56:11 | cuda
18:56:11 | vaapi
18:56:11 | qsv
18:56:11 | drm
18:56:11 | opencl
18:56:11 | vulkan
18:56:11 | Found 0 devices
18:56:11 | call arecord -l
18:56:11 | Preloading Images
18:56:11 | Turn Server accepting connections at 172.17.0.2:3478 using range 50000-50010)
18:56:11 | MQTT not configured or enabled
18:56:11 | Loaded 0 cameras and 0 mics from /agent/Media/XML/objects.xml
18:56:11 | Using default port (8090)
18:56:11 | Local Server running on port 8090
18:56:11 | Access locally at http://172.17.0.2:8090
18:56:11 | IP addresses:
18:56:11 | 172.17.0.2
18:56:11 | Accepting http connections at http://*:8090/
18:56:11 | found 172.17.0.2
18:56:11 | found 172.17.0.1
18:56:12 | Running at http://localhost:8090
18:56:12 | ZeroConf: ZeroConf Service Started
18:56:15 | Using comms server https://agentrtc.azurewebsites.net
18:56:19 | Connecting service (https://agentrtc.azurewebsites.net)
18:56:24 | Connected to SignalR. Waiting for hello...
18:56:26 | SignalR: Services Connected
18:56:26 | Welcome: Fitap
18:56:26 | Connected - access this server remotely via https://www.ispyconnect.com/
18:57:03 | rtc:added servers
18:57:03 | rtc:onsuccessanswer
18:57:03 | ICE state: Checking
18:57:03 | ICE state: Connected
18:57:03 | rtc:ondatachannelstatechange (serverdata: Open)
18:57:03 | rtc: opened serverdata
18:57:03 | rtc:ondatachannelstatechange (talk: Open)
18:57:03 | rtc: opened talk
18:59:17 | Port 80: open
18:59:19 | Discovery (GetServices): ONVIF v1.0
18:59:19 | Discovery: Using Media Version 1
18:59:19 | Discovery: Media available at http://192.168.0.31/onvif/media_service
18:59:19 | Discovery: PTZ available at http://192.168.0.31/onvif/ptz_service
18:59:19 | Discovery: Events available at http://192.168.0.31/onvif/event_service
18:59:22 | ONVIF Presets: Presets not supported
18:59:22 | Error: ONVIF Presets: Presets not supported
18:59:22 | ONVIF Presets: Presets not supported
18:59:22 | Error: ONVIF Presets: Presets not supported
18:59:22 | Port 554: open
18:59:32 | Port 8090: closed
18:59:34 | Port 1935: closed
18:59:48 | Enable: Camera 1: Enabled
18:59:48 | Discovery (GetServices): Camera 1: ONVIF v1.0
18:59:48 | Discovery: Camera 1: Using Media Version 1
18:59:48 | Discovery: Camera 1: Media available at http://192.168.0.31/onvif/media_service
18:59:48 | Discovery: Camera 1: PTZ available at http://192.168.0.31/onvif/ptz_service
18:59:48 | Discovery: Camera 1: Events available at http://192.168.0.31/onvif/event_service
18:59:50 | ONVIF Presets: Camera 1: Presets not supported
18:59:50 | Error: ONVIF Presets: Camera 1: Presets not supported
18:59:51 | open (live): Camera 1: Trying UDP
18:59:51 | Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:51 | Error: Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:51 | Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
18:59:51 | Error: Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
18:59:51 | Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:51 | SourceError:  Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader: Camera 1:  OPEN_INPUT:Invalid data found when processing input
18:59:51 | Error:  SourceError: Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader:  Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:51 | SourcePlayingFinished: Camera 1: Playing finished (SourceError)
18:59:51 | Reconnect: Camera 1: Reconnecting in 2s
18:59:53 | intervals: Camera 1: Reconnecting
18:59:53 | intervals: Camera 1: Reconnect Exit
18:59:54 | Discovery (GetServices): Camera 1: ONVIF v1.0
18:59:54 | Discovery: Camera 1: Using Media Version 1
18:59:54 | Discovery: Camera 1: Media available at http://192.168.0.31/onvif/media_service
18:59:54 | Discovery: Camera 1: PTZ available at http://192.168.0.31/onvif/ptz_service
18:59:54 | Discovery: Camera 1: Events available at http://192.168.0.31/onvif/event_service
18:59:55 | ONVIF Presets: Camera 1: Presets not supported
18:59:55 | Error: ONVIF Presets: Camera 1: Presets not supported
18:59:55 | open (live): Camera 1: Trying TCP
18:59:55 | Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:55 | Error: Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:55 | Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
18:59:55 | Error: Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
18:59:55 | Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:55 | SourceError:  Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader: Camera 1:  OPEN_INPUT:Invalid data found when processing input
18:59:55 | Error:  SourceError: Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader:  Camera 1: OPEN_INPUT:Invalid data found when processing input
18:59:55 | SourcePlayingFinished: Camera 1: Playing finished (SourceError)
18:59:55 | Reconnect: Camera 1: Reconnecting in 5s
19:00:01 | intervals: Camera 1: Reconnecting
19:00:01 | intervals: Camera 1: Reconnect Exit
19:00:02 | Discovery (GetServices): Camera 1: ONVIF v1.0
19:00:02 | Discovery: Camera 1: Using Media Version 1
19:00:02 | Discovery: Camera 1: Media available at http://192.168.0.31/onvif/media_service
19:00:02 | Discovery: Camera 1: PTZ available at http://192.168.0.31/onvif/ptz_service
19:00:02 | Discovery: Camera 1: Events available at http://192.168.0.31/onvif/event_service
19:00:04 | ONVIF Presets: Camera 1: Presets not supported
19:00:04 | Error: ONVIF Presets: Camera 1: Presets not supported
19:00:04 | open (live): Camera 1: Trying UDP
19:00:04 | Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:04 | Error: Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:04 | Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
19:00:04 | Error: Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
19:00:04 | Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:04 | SourceError:  Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader: Camera 1:  OPEN_INPUT:Invalid data found when processing input
19:00:04 | Error:  SourceError: Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader:  Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:04 | SourcePlayingFinished: Camera 1: Playing finished (SourceError)
19:00:04 | Reconnect: Camera 1: Reconnecting in 10s
19:00:14 | intervals: Camera 1: Reconnecting
19:00:14 | intervals: Camera 1: Reconnect Exit
19:00:15 | Discovery (GetServices): Camera 1: ONVIF v1.0
19:00:15 | Discovery: Camera 1: Using Media Version 1
19:00:15 | Discovery: Camera 1: Media available at http://192.168.0.31/onvif/media_service
19:00:15 | Discovery: Camera 1: PTZ available at http://192.168.0.31/onvif/ptz_service
19:00:15 | Discovery: Camera 1: Events available at http://192.168.0.31/onvif/event_service
19:00:17 | ONVIF Presets: Camera 1: Presets not supported
19:00:17 | Error: ONVIF Presets: Camera 1: Presets not supported
19:00:17 | open (live): Camera 1: Trying TCP
19:00:17 | Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:17 | Error: Camera 1: DoStart: Camera 1: DoStart: Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:17 | Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
19:00:17 | Error: Camera 1: DoStart:    at CoreLogic.Sources.Combined.MediaStream.DoStart()
19:00:17 | Reader: Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:17 | SourceError:  Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader: Camera 1:  OPEN_INPUT:Invalid data found when processing input
19:00:17 | Error:  SourceError: Camera 1: CoreLogic.Sources.Combined.MediaStream:Reader:  Camera 1: OPEN_INPUT:Invalid data found when processing input
19:00:17 | SourcePlayingFinished: Camera 1: Playing finished (SourceError)
19:00:17 | Reconnect: Camera 1: Reconnecting in 30s
19:00:22 | Discovery (GetServices): ONVIF v1.0
19:00:22 | Discovery: Using Media Version 1
19:00:22 | Discovery: Media available at http://192.168.0.31/onvif/media_service
19:00:22 | Discovery: PTZ available at http://192.168.0.31/onvif/ptz_service
19:00:22 | Discovery: Events available at http://192.168.0.31/onvif/event_service

**Comment by ispysoftware** [comment]
Hello this response generally means the camera has responded with an error page or something other than a video stream. Try the URL in a web browser and see what response you get - possibly the username/ password/ channel is incorrect or the password might contain special characters that are breaking the URL.

**Comment by Fitap** [comment]
Everything ok for now, it was a misconfiguration.
I'm sorry.

**Reply by ManiekPrz** [reply]
What setup links ports etc you had for imou camera - I'm trying to setup Imou Cruiser Dual with no luck :( . I have port, admin password and any of tools is able to connect - no idea is it onvif or any other standard to follow. Nor iSpy nor anycam can detect it :(
